safe-transaction-client/src/lib.rs

```diff
@@ -0,0 +1,343 @@
+            }
+            e => Error::Ureq(e),
+        }
+    }
+}
+
+impl From<io::Error> for Error {
+    fn from(e: io::Error) -> Self {
+        Self::IO(e)
+    }
+}
+
+impl From<SignatureError> for Error {
+    fn from(e: SignatureError) -> Self {
+        Self::SignatureError(e)
+    }
+}
+
+impl<E> From<E> for Error<E> {
+    fn from(e: E) -> Self {
+        Self::Signature(e)
+    }
+}
+
+pub type Result<T = (), E = Error> = result::Result<T, E>;
+
+#[derive(Debug, Clone, Copy)]
+#[repr(u8)]
+pub enum Operation {
+    Call,
+    DelegateCall,
+}
+
+pub struct Client<'a> {
+    agent: ureq::Agent,
+    transactions_api: &'a str,
+}
+
+pub struct Safe<'a> {
+    client: &'a Client<'a>,
+    safe_address: Address,
+}
+
+#[derive(Debug)]
+// All fields are public, so you may construct it directly or update the nonce,
+// prices or refund address as desired.
+pub struct SafeTx {
+    pub safe_address: Address,
+    pub to: Address,
+    pub value: U256,
+    pub data: Bytes,
+    pub operation: Operation,
+    pub nonce: U256,
+}
+
+#[derive(Debug)]
+pub struct SignedSafeTx {
+    inner: SafeTx,
+    sender: Address,
+    safe_tx_hash: TxHash,
+    signature: Signature,
+}
+
+impl Client<'_> {
+    pub fn new(transactions_api: &str) -> Client {
+        Client {
+            agent: ureq::Agent::new(),
+            transactions_api,
+        }
+    }
+
+    pub fn with_safe(&self, safe_address: Address) -> Safe {
+        Safe {
+            client: self,
+            safe_address,
+        }
+    }
+}
+
+impl Safe<'_> {
+    pub fn create_transaction(
+        &self,
+        to: Address,
+        value: U256,
+        data: Bytes,
+        operation: Operation,
+    ) -> Result<SafeTx> {
+        #[derive(serde::Deserialize)]
+        struct SafeResponse {
+            nonce: u64,
+        }
+
+        let SafeResponse { nonce, .. } = self
+            .client
+            .agent
+            .get(&format!(
+                "{}/v1/safes/{}/",
+                self.client.transactions_api,
+                to_checksum(&self.safe_address, None),
+            ))
+            .call()?
+            .into_json()?;
+
+        Ok(SafeTx {
+            safe_address: self.safe_address,
+            to,
+            value,
+            data,
+            nonce: U256::from(nonce),
+            operation,
+        })
+    }
+
+    pub fn propose(&self, signed_safe_tx: SignedSafeTx) -> Result {
+        let SafeTx {
+            to,
+            value,
+            data,
+            nonce,
+            operation,
+            ..
+        } = signed_safe_tx.inner;
+        self.client
+            .agent
+            .post(&format!(
+                "{}/v1/safes/{}/multisig-transactions/",
+                self.client.transactions_api,
+                to_checksum(&self.safe_address, None),
+            ))
+            .send_json(ureq::json!({
+                "to": to_checksum(&to, None),
+                "value": format!("{}", value),
```

Comment:
Isn't this the equivalent of `value.to_string()` or `value.display()`?

Reply:
`U256::to_string()` yes, there is no `U256::display`.